components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -280,4 +287,70 @@ public void addBotDataEmailConfiguration(String email) throws APIManagementExcep
     public void deleteBotDataEmailList(String uuid) throws APIManagementException, SQLException {
         apiMgtDAO.deleteBotDataEmailList(uuid);
     }
+
+    public APICategory addCategory(APICategory category, String userName) throws APIManagementException {
+        int tenantID = APIUtil.getTenantId(userName);
+        if (isCategoryNameExists(category.getName(), null, tenantID)) {
+            APIUtil.handleException("Category with name '" + category.getName() + "' already exists");
+        }
+        return apiMgtDAO.addCategory(tenantID, category);
+    }
+
+    public void updateCategory(APICategory apiCategory) throws APIManagementException {
+        apiMgtDAO.updateCategory(apiCategory);
+    }
+
+    public void deleteCategory(String categoryID, String username) throws APIManagementException {
+        APICategory category = getAPICategoryByID(categoryID);
+        int attchedAPICount = isCategoryAttached(category, username);
+        if ( attchedAPICount > 0) {
+            APIUtil.handleException("Unable to delete the category. It is attached to API(s)");
+        }
+        apiMgtDAO.deleteCategory(categoryID);
+    }
+
+    public List<APICategory> getAllAPICategoriesOfTenant(int tenantId) throws APIManagementException {
+        return apiMgtDAO.getAllCategories(tenantId);
+    }
+
+    public List<APICategory> getAllAPICategoriesOfTenantForAdminListing(String username) throws APIManagementException{
+        int tenantID = APIUtil.getTenantId(username);
+        List<APICategory> categories = getAllAPICategoriesOfTenant(tenantID);
+        if (categories.size() > 0) {
+            for (APICategory category : categories) {
+                int length = isCategoryAttached(category, username);
+                category.setNumberOfAPIs(length);
+            }
+
```

Comment:
extra space

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/8077/commits/e646d5a8cbf82d6fb2ba2cd090e252a4dfaca540